core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -175,18 +197,114 @@ public void verifyAddPage()
         public enum TestType
         {
             BIGINT(BigintType.BIGINT, 5000),
+            BIGINT_SKEWED_HASH(BigintType.BIGINT, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return page(
+                            positionCount,
+                            types.size(),
+                            () -> createRandomBlockForType(BigintType.BIGINT, positionCount, nullRate),
+                            createRandomLongsBlock(positionCount, 2));
+                }
+            },
             DICTIONARY_BIGINT(BigintType.BIGINT, 3000) {
                 @Override
                 public Page createPage(List<Type> types, int positionCount, float nullRate)
                 {
-                    return createRandomDictionaryPage(types, POSITION_COUNT, nullRate);
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
                 }
             },
             RLE_BIGINT(BigintType.BIGINT, 3000) {
                 @Override
                 public Page createPage(List<Type> types, int positionCount, float nullRate)
                 {
-                    return createRandomRlePage(types, POSITION_COUNT, nullRate);
+                    return createRandomRlePage(types, positionCount, nullRate);
+                }
+            },
+            BIGINT_PARTITION_BIGINT_20_PERCENT(BigintType.BIGINT, 3000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return page(
+                            positionCount,
+                            types.size(),
+                            () -> createRandomBlockForType(BigintType.BIGINT, positionCount, nullRate),
+                            createLongsBlock(LongStream.range(0, positionCount)
+                                    .mapToObj(value -> value % (positionCount / 5))
+                                    .collect(toImmutableList())));
+                }
+            },
+            DICTIONARY_PARTITION_BIGINT_20_PERCENT(BigintType.BIGINT, 3000) {
```

Comment:
`...` -> `BIGINT_DICTIONARY_PARTITION_CHANNEL_20_PERCENT` I think it's more explicit (english) this way